knotx-core/src/main/java/io/knotx/dataobjects/Fragment.java

```diff
@@ -36,38 +36,56 @@
   private static final String KNOTS_KEY = "_KNOTS";
   private static final String CONTENT_KEY = "_CONTENT";
   private static final String CONTEXT_KEY = "_CONTEXT";
+  private static final String FAILED_KEY = "_FAILED";
+  private static final String FAILURE_DETAILS_KEY = "_FAILEURE_DETAILS";
+  private static final String FALLBACK_KEY = "_FALLBACK";
+
 
   private final List<String> knots;
   private final JsonObject context;
   private String content;
+  private boolean failed;
+  private String failureDetails;
```

Comment:
I think about the solution that instead of `failureDetails` can provide more details about the flow. We could extend `knots` list with such details as:
- knotIdentifier
- processed: SUCCESS, FAILURE, UNPROCESSED
- errors:
   - error code
   - error message

So if for example Data Bridge would not be able to call 2 services we  would get:
- knotIndetifier: dataBridge
- processed: FAILURE
- errors [ {error_from_service_1}, {error_from_service_2} ]

Reply:
changes implemented here, unfortunately github is not working that great right now so they are not visible in the PR: https://github.com/scypio/knotx/commit/66be8f930a7a15140ff21c6bae31e2dcb33dc47d

Reply:
added Knot object to Fragment with knot name / status and errors information